exporters/otlp/otlplog/otlploggrpc/internal/transform/log_test.go

```diff
@@ -226,6 +227,81 @@ func TestResourceLogs(t *testing.T) {
 	assert.Equal(t, want, ResourceLogs(records))
 }
 
+func TestResourceLogsPerResource(t *testing.T) {
+	res1 := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		attribute.KeyValue{
+			Key:   "service.name",
+			Value: attribute.StringValue("service1"),
+		},
+	)
+	res2 := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		attribute.KeyValue{
+			Key:   "service.name",
+			Value: attribute.StringValue("service2"),
+		},
+	)
+
+	scope := instrumentation.Scope{
+		Name:      "test/code/path1",
+		Version:   "v0.2.0",
+		SchemaURL: semconv.SchemaURL,
+	}
+	records = func() []log.Record {
+		var out []log.Record
+
+		out = append(out, logtest.RecordFactory{
+			Timestamp:         ts,
+			ObservedTimestamp: obs,
+			Body:              bodyA,
+			Attributes:        []api.KeyValue{alice},
+			Resource:          res1,
+		}.NewRecord())
+
+		out = append(out, logtest.RecordFactory{
+			Timestamp:         ts,
+			ObservedTimestamp: obs,
+			Body:              bodyB,
+			Attributes:        []api.KeyValue{bob},
+			Resource:          res1,
+		}.NewRecord())
+
+		out = append(out, logtest.RecordFactory{
+			Timestamp:         ts,
+			ObservedTimestamp: obs,
+			Body:              bodyB,
+			Attributes:        []api.KeyValue{bob},
+			Resource:          res2,
+		}.NewRecord())
+
+		out = append(out, logtest.RecordFactory{
+			Timestamp:            ts,
+			ObservedTimestamp:    obs,
+			Body:                 bodyB,
+			Attributes:           []api.KeyValue{alice},
+			Resource:             res2,
+			InstrumentationScope: &scope,
+		}.NewRecord())
```

Comment:
The log record bodies, attributes, etc. are set, but later they are not verified in the assertion.

I suggest removing setting fields which are not important for the unit test or make the assertions more strict.

Reply:
Makes sense @pellared , I will make the necessary changes.